File: VB/CS/E911.Win/Controls/WinCustomUserControl.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Utils
Imports E911.Module.Editors
Imports System.Collections.Generic

Namespace E911.Module.Win.Controls
	Partial Public Class WinCustomUserControl
		Inherits DevExpress.XtraEditors.XtraUserControl
		Implements IXpoSessionAwareControl
		Public Sub New()
			InitializeComponent()
		End Sub
		Protected Overrides Sub OnLoad(ByVal e As EventArgs)
			MyBase.OnLoad(e)
			If (Not DesignMode) Then
				UpdateDataSource()
			End If
		End Sub
		Private Sub UpdateDataSource()
			Dim session As Session = (CType(Me, IXpoSessionAwareControl)).Session
			Guard.ArgumentNotNull(session, "session")
			gridControl1.DataSource = New XPCollection(Of DevExpress.Persistent.BaseImpl.Task)(session)
			gridControl1.ForceInitialize()
		End Sub
		Private privateSession As Session
		Private Property Session() As Session Implements IXpoSessionAwareControl.Session
			Get
				Return privateSession
			End Get
			Set(ByVal value As Session)
				privateSession = value
			End Set
		End Property
	End Class
End Namespace

File: CS/E911.Win/Controls/WinCustomUserControl.cs
using System;
using DevExpress.Xpo;
using DevExpress.Utils;
using System.Collections;
using E911.Module.Editors;
using System.Collections.Generic;
using DevExpress.Data.Filtering;

namespace E911.Module.Win.Controls {
    /// <summary>
    /// This is a custom WinForms user control that displays persistent data received from XPO.
    /// You do not need to implement the IXpoSessionAwareControl interface if your control gets data from other sources or does not require data at all.
    /// </summary>
    public partial class WinCustomUserControl : DevExpress.XtraEditors.XtraUserControl, IXpoSessionAwareControl {
        public WinCustomUserControl() {
            InitializeComponent();
        }
        void IXpoSessionAwareControl.UpdateDataSource(Session session) {
            Guard.ArgumentNotNull(session, "session");
            Type persistentDataType = typeof(DevExpress.Persistent.BaseImpl.Task);
            IList persistentData = new XPCollection(session, persistentDataType, CriteriaOperator.Parse("Status = 'NotStarted'"));
            gridControl1.DataSource = persistentData;
            gridControl1.ForceInitialize();
        }
    }
}
